frontend.go

```diff
@@ -107,11 +107,8 @@ func (f *Frontend) load(ctx context.Context, key string, loader HTTPLoader) (Ent
 
 	defer span.End()
 
-	data, err := f.Do(key, func() (res interface{}, resultErr error) {
-		ctx, fetchRoutineSpan := trace.StartSpan(
-			newContextWithSpan,
-			"flamingo/httpcache/fetchRoutine",
-		)
+	data, err, _ := f.Do(key, func() (res interface{}, resultErr error) {
+		ctx, fetchRoutineSpan := trace.StartSpan(newContextWithSpan, "flamingo/httpcache/fetchRoutine")
 		fetchRoutineSpan.Annotate(nil, key)
 		defer fetchRoutineSpan.End()
```

Comment:
should we end span directly at the end of fetch routine?

Reply:
Good point, marked it as a sub span now looks like this:
![2023-07-03 at 10 08](https://github.com/i-love-flamingo/httpcache/assets/3203968/f1dc81d1-9300-41c6-8bb3-12b002ce736c)